cmd/kube-proxy/app/server.go

```diff
@@ -393,15 +394,18 @@ type ProxyServer struct {
 // createClients creates a kube client and an event client from the given config and masterOverride.
 // TODO remove masterOverride when CLI flags are removed.
 func createClients(config componentconfig.ClientConnectionConfiguration, masterOverride string) (clientset.Interface, v1core.EventsGetter, error) {
+	var kubeConfig *rest.Config
+	var err error
+
 	if len(config.KubeConfigFile) == 0 && len(masterOverride) == 0 {
-		glog.Warningf("Neither --kubeconfig nor --master was specified. Using default API client. This might not work.")
```

Comment:
Do you want to log that it's using in-cluster config?

Reply:
+1 on logging why it is falling back to in cluster config. LGTM otherwise

Reply:
Sounds good. Added a log line for this.